articles/speech-command-recognition-with-tensorflow-and-react/index.md

```diff
@@ -0,0 +1,166 @@
+<!--more-->
+In this tutorial, we will learn how to perform speech command recognition in real-time and demonstrate the result of predictions. Then, we are going to load the speech-commands model provided by TensorFlow.org. Feeding the audio data to the model neural network, we are going to listen to the audio and make the predictions. Lastly, we are also going to improve the accuracy recognition by applying the argmax function to our model.
+
+### What we will cover in this tutorial...
+
+- To access the audio data stream from the browser microphone.
+- To precise, the speech commands using the pre-trained speech-commands model from TensorFlow.
+- To improve the overall prediction/recognition probability by using the argmax function.
+
+*Let's get started!*
+
+### Create React Project
+First, we are going to create a new React App project. For that, we need to run the following command in the required local directory:
+
+```jsx
+npx create-react-app speech_command
+```
+
+Then, we need to open the project in our code editor IDE (VScode) as shown in the screenshot below:
+
+![create new react app](/engineering-education/speech-command-recognition-with-tensorflow-and-react/1-create-new-react-app.png)
+
+Now, we can execute the command  `yarn start` or `npm start` to run the project on the port 3000 of our [localhost](http://localhost):
+
+![preview react app](/engineering-education/speech-command-recognition-with-tensorflow-and-react/2-preview-react-app.png)
+
+### Installing Tensorflow.JS & Speech Command Recognition
+
+Now, we are going to install the main TensorFlow.JS package along with the speech commands recognition package provided by TensorFlow.org. for installation package, we need to execute the following command in our project commmand line or terminal:
+
+```bash
+yarn add @tensorflow/tfjs @tensorflow-models/speech-commands
+```
+
+- @tensorflow/tfjs: It is the core Tensorflow package based on JavaScript.
+- @tensorflow-models/speech-commands: It is a Tensorflow models that enables us to perform recognition of spoken commands.
+
+### Importing Dependencies
+
+After the successful installation of the required packages. we need to import them in the App.js file of our React project as shown in the code snippet below:
+
+```jsx
+import * as tf from "@tensorflow/tfjs"
+import * as speech from "@tensorflow-models/speech-commands"
+```
+
+### Setting Up the Application States
+
+Then, we need to define the state variable for model, action, and labels using the `useState` hook as shown in the code snippet below:
```

Comment:
Please briefly explain how the useState hook works. This might be useful for a beginner